```
FAIL: Swift(linux-x86_64) :: Concurrency/Runtime/async_task_priority_current.swift (555 of 16924)
******************** TEST 'Swift(linux-x86_64) :: Concurrency/Runtime/async_task_priority_current.swift' FAILED ********************
Script:
--
: 'RUN: at line 1';   rm -rf "/home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp" && mkdir -p "/home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp" && /home/build-user/build/buildbot_linux/swift-linux-x86_64/bin/swiftc -target x86_64-unknown-linux-gnu -toolchain-stdlib-rpath  -module-cache-path /home/build-user/build/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4  -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0' -Xfrontend -define-availability -Xfrontend 'SwiftStdlib 5.10:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999'  -module-cache-path /home/build-user/build/buildbot_linux/swift-linux-x86_64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache /home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift  -Xfrontend -disable-availability-checking -I /home/build-user/swift-corelibs-libdispatch -I /home/build-user/build/buildbot_linux/libdispatch-linux-x86_64/src/swift/swift -L /home/build-user/build/buildbot_linux/libdispatch-linux-x86_64 -vfsoverlay /home/build-user/build/buildbot_linux/libdispatch-linux-x86_64/dispatch-vfs-overlay.yaml -parse-as-library -o /home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp/a.out -module-name main  && echo /home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp/a.out && /usr/bin/env LD_LIBRARY_PATH='/home/build-user/build/buildbot_linux/swift-linux-x86_64/lib/swift/linux:/home/build-user/build/buildbot_linux/libdispatch-linux-x86_64'  /home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp/a.out | /usr/bin/python3.8 /home/build-user/swift/utils/PathSanitizingFileCheck --allow-unused-prefixes --sanitize BUILD_DIR=/home/build-user/build/buildbot_linux/swift-linux-x86_64 --sanitize SOURCE_DIR=/home/build-user/swift --use-filecheck /home/build-user/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck  --dump-input=always /home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift
--
Exit Code: 1

Command Output (stdout):
--
/home/build-user/build/buildbot_linux/swift-linux-x86_64/test-linux-x86_64/Concurrency/Runtime/Output/async_task_priority_current.swift.tmp/a.out

--
Command Output (stderr):
--
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:14:25: warning: conformance of 'TaskPriority' to protocol 'CustomStringConvertible' was already stated in the type's module '_Concurrency'
extension TaskPriority: CustomStringConvertible {
                        ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:15:14: note: property 'description' will not be used to satisfy the conformance to 'CustomStringConvertible'
  public var description: String {
             ^
_Concurrency.TaskPriority:2:11: note: 'TaskPriority' declares conformance to protocol 'CustomStringConvertible' here
extension TaskPriority : CustomStringConvertible {
          ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:73:9: warning: expression of type 'Task<(), Never>' is unused
  await x
        ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:74:9: warning: expression of type 'Task<(), Never>' is unused
  await z
        ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:59:9: warning: no 'async' operations occur within 'await' expression
    _ = await z // waiting on `z`, but it won't complete since we're also background
        ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:65:5: warning: no 'async' operations occur within 'await' expression
    await x // escalates x, which waits on z, so z also escalates
    ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:73:3: warning: no 'async' operations occur within 'await' expression
  await x
  ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:74:3: warning: no 'async' operations occur within 'await' expression
  await z
  ^
/home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift:23:60: error: CHECK: expected string not found in input
 print("main priority: \(Task.currentPriority)") // CHECK: main priority: TaskPriority(rawValue: [[#MAIN_PRIORITY:]])
                                                           ^
<stdin>:1:1: note: scanning from here
main priority: TaskPriority.medium
^

Input file: <stdin>
Check file: /home/build-user/swift/test/Concurrency/Runtime/async_task_priority_current.swift

-dump-input=help explains the following input dump.

Input was:
<<<<<<
          1: main priority: TaskPriority.medium 
check:23     X~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ error: no match found
          2: a1: TaskPriority.medium 
check:23     ~~~~~~~~~~~~~~~~~~~~~~~~
          3: a2: TaskPriority.high 
check:23     ~~~~~~~~~~~~~~~~~~~~~~
          4: a3: TaskPriority.medium 
check:23     ~~~~~~~~~~~~~~~~~~~~~~~~
          5: default done 
check:23     ~~~~~~~~~~~~~
>>>>>>

--

********************
```

--- a/test/Concurrency/Runtime/async_task_priority_current.swift
+++ b/test/Concurrency/Runtime/async_task_priority_current.swift
@@ -10,17 +10,10 @@
 
 import Dispatch
 
-// Work around the inability of older Swift runtimes to print a task priority.
-extension TaskPriority: CustomStringConvertible {
-  public var description: String {
-    "TaskPriority(rawValue: \(rawValue))"
-  }
-}
-
 @available(SwiftStdlib 5.1, *)
 @main struct Main {
   static func main() async {
-    print("main priority: \(Task.currentPriority)") // CHECK: main priority: TaskPriority(rawValue: [[#MAIN_PRIORITY:]])
+    print("main priority: \(Task.currentPriority)") // CHECK: main priority: TaskPriority.medium
     await test_detach()
     await test_multiple_lo_indirectly_escalated()
   }
@@ -29,18 +22,18 @@ extension TaskPriority: CustomStringConvertible {
 @available(SwiftStdlib 5.1, *)
 func test_detach() async {
   let a1 = Task.currentPriority
-  print("a1: \(a1)") // CHECK: a1: TaskPriority(rawValue: [[#MAIN_PRIORITY]])
+  print("a1: \(a1)") // CHECK: a1: TaskPriority.medium
 
   // Note: remember to detach using a higher priority, otherwise a lower one
   // might be escalated by the get() and we could see `default` in the detached
   // task.
   await detach(priority: .userInitiated) {
     let a2 = Task.currentPriority
-    print("a2: \(a2)") // CHECK: a2: TaskPriority(rawValue: 25)
+    print("a2: \(a2)") // CHECK: a2: TaskPriority.high
   }.get()
 
   let a3 = Task.currentPriority
-  print("a3: \(a3)") // CHECK: a3: TaskPriority(rawValue: [[#MAIN_PRIORITY]])
+  print("a3: \(a3)") // CHECK: a3: TaskPriority.medium
 }
 
 @available(SwiftStdlib 5.1, *)

--- a/test/Concurrency/Runtime/custom_executors_complex_equality_crash.swift
+++ b/test/Concurrency/Runtime/custom_executors_complex_equality_crash.swift
@@ -1,4 +1,4 @@
-// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking -parse-as-library)
+// RUN: %target-run-simple-swift( -Xfrontend -disable-availability-checking %import-libdispatch -parse-as-library)
 
 // REQUIRES: concurrency
 // REQUIRES: executable_test

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -1643,15 +1643,17 @@ elif (run_os in ['linux-gnu', 'linux-gnueabihf', 'freebsd', 'openbsd', 'windows-
     libdispatch_artifact_dir = config.libdispatch_build_path
     libdispatch_swift_module_dir = make_path(libdispatch_artifact_dir, 'src', 'swift', 'swift')
     libdispatch_source_dir = make_path(config.swift_src_root, os.pardir, 'swift-corelibs-libdispatch')
+    libdispatch_vfs_yaml =  make_path(libdispatch_artifact_dir, 'dispatch-vfs-overlay.yaml')
     libdispatch_artifacts = [
+        libdispatch_vfs_yaml,
         make_path(libdispatch_artifact_dir, 'libdispatch.so'),
         make_path(libdispatch_artifact_dir, 'libswiftDispatch.so'),
         make_path(libdispatch_swift_module_dir, 'Dispatch.swiftmodule')]
     if (all(os.path.exists(p) for p in libdispatch_artifacts)):
         config.available_features.add('libdispatch')
         config.libdispatch_artifact_dir = libdispatch_artifact_dir
-        config.import_libdispatch = ('-I %s -I %s -L %s'
-            % (libdispatch_source_dir, libdispatch_swift_module_dir, libdispatch_artifact_dir))
+        config.import_libdispatch = ('-I %s -I %s -L %s -vfsoverlay %s'
+            % (libdispatch_source_dir, libdispatch_swift_module_dir, libdispatch_artifact_dir, libdispatch_vfs_yaml))
 
     libdispatch_static_artifact_dir = os.path.join(config.libdispatch_static_build_path, 'lib')
     libdispatch_static_artifacts = [

